pkg/agent/packetcapture/packetcapture_controller_linux.go

```diff
@@ -0,0 +1,884 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package packetcapture
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/gopacket/gopacket"
+	"github.com/gopacket/gopacket/layers"
+	"github.com/gopacket/gopacket/pcapgo"
+	"github.com/spf13/afero"
+	"golang.org/x/time/rate"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/client-go/util/workqueue"
+	klog "k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/packetcapture/bpf"
+	"antrea.io/antrea/pkg/agent/util"
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	clientsetversioned "antrea.io/antrea/pkg/client/clientset/versioned"
+	crdinformers "antrea.io/antrea/pkg/client/informers/externalversions/crd/v1alpha1"
+	crdlisters "antrea.io/antrea/pkg/client/listers/crd/v1alpha1"
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+	"antrea.io/antrea/pkg/util/env"
+	"antrea.io/antrea/pkg/util/ftp"
+)
+
+type storageProtocolType string
+
+const (
+	sftpProtocol storageProtocolType = "sftp"
+)
+
+const (
+	controllerName               = "PacketCaptureController"
+	resyncPeriod   time.Duration = 0
+
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+
+	defaultWorkers = 2
+
+	// defines how many capture request we can handle concurrently.
+	maxConcurrentCaptures = 4
+	// waiting captures will be marked as Pending unitl it can be processed.
+	maxWaitingCaptures = 32
+
+	// reason for timeout
+	captureTimeoutReason   = "PacketCapture timeout"
+	defaultTimeoutDuration = time.Second * time.Duration(60)
+
+	captureStatusUpdatePeriod = 10 * time.Second
+
+	// PacketCapture uses a dedicated Secret object to store authentication information for a file server.
+	// #nosec G101
+	fileServerAuthSecretName = "antrea-packetcapture-fileserver-auth"
+
+	// Max packet size for pcap capture.
+	maxSnapshotBytes = 65536
+)
+
+type packetCapturePhase string
+
+const (
+	packetCapturePhaseUnknown   packetCapturePhase = ""
+	packetCapturePhasePending   packetCapturePhase = "Pending"
+	packetCapturePhaseRunning   packetCapturePhase = "Running"
+	packetCapturePhaseCompleted packetCapturePhase = "Completed"
+)
+
+var (
+	packetDirectory = filepath.Join(os.TempDir(), "antrea", "packetcapture", "packets")
+	defaultFS       = afero.NewOsFs()
+)
+
+type packetCaptureState struct {
+	// name is the PacketCapture name.
+	name string
+	// capturedPacketsNum record how many packets have been captured. Due to the RateLimiter,
+	// this maybe not be the real-time data.
```

Comment:
```suggestion
	// capturedPacketsNum records how many packets have been captured. Due to the RateLimiter,
	// this may not be the real-time data.
```